### PR TITLE
 TKO-53 BE: Student Filter by Major

### PR DESCRIPTION
**Pull request by nixy97**
# Description
Create a Route to filter students based on major, given a major

# Testing
on this branch go on http://localhost:8080/students/major/compsci
it should give an empty array

# Installation
composer install

### CODE REVIEW

File: app/Http/Controllers/AdminController.php [hunk 1]
```diff
@@ -5,20 +5,28 @@
...
 use Mail;
 use App\Mail\InviteStudent;
+use App\Contracts\StudentInfoContract;
 
 class AdminController extends BaseController
 {
     private $adminRetriever;
 
-    public function __construct(AdminContract $adminContract)
+    public function __construct(StudentInfoContract $studentinfoContract, AdminContract $adminContract)
```
Comment (cbenavidesdev) [inline app/Http/Controllers/AdminController.php]: I would separate these into multiple lines like
public function __construct(
                 StudentInfoContract $studentinfoContract,
                 AdminContract $adminContract
               )

File: app/Providers/StudentInfoServiceProvider.php [hunk 2]
```diff
@@ -19,6 +19,8 @@ public function boot()
 
     public function register()
     {
-        $this->app->bind('App\Contracts\StudentInfoContract', 'App\Services\StudentInfoService');
+        $this->app->bind(
+            'App\Contracts\StudentInfoContract',
+            'App\Services\StudentInfoService');
```
Comment (carlosbenavides123) [inline app/Providers/StudentInfoServiceProvider.php]: clean this up

File: app/Models/StudentInfo.php [hunk 3]
```diff
@@ -26,4 +27,7 @@ class StudentInfo extends Model
         'academic_interest'
     ];
 
+    public function users(){
+        return $this->belongsTo('App\User', 'id', 'user_id');
+    }
```
Comment (cbenavidesdev) [inline app/Models/StudentInfo.php]: consider moving App\Users into the models folder when you get the chance

**Review by cbenavidesdev — APPROVED** [review]
👍

File: app/Models/StudentInfo.php (hunk 3 above)
Comment (MikkalMc) [reply app/Models/StudentInfo.php]: agreed

**Review by MikkalMc — APPROVED** [review]
Looks good 👍

**Review by luisjg — APPROVED** [review]
👍

**Pull request merged**